File: examples/typescript/send-email.ts
/**
 * Send an email (and reply) from an Inkbox mailbox.
 *
 * Usage:
 *   INKBOX_API_KEY=ApiKey_... MAILBOX_ADDRESS=agent@inkboxmail.com npx ts-node send-email.ts
 */

import { InkboxMail } from "../../typescript/src/client.js";

const client = new InkboxMail({ apiKey: process.env.INKBOX_API_KEY! });
const mailboxAddress = process.env.MAILBOX_ADDRESS!;

// Send a new email
const sent = await client.messages.send(mailboxAddress, {
  to: ["recipient@example.com"],
  subject: "Hello from Inkbox",
  bodyText: "Hi there! This message was sent via the Inkbox SDK.",
  bodyHtml: "<p>Hi there! This message was sent via the <strong>Inkbox SDK</strong>.</p>",
});
console.log(`Sent message ${sent.id}  subject="${sent.subject}"`);

// Reply to that message (threads it automatically)
const reply = await client.messages.send(mailboxAddress, {
  to: ["recipient@example.com"],
  subject: `Re: ${sent.subject}`,
  bodyText: "Following up on my previous message.",
  inReplyToMessageId: sent.id,
});
console.log(`Sent reply ${reply.id}`);

File: examples/python/send_email.py
"""
Send an email (and reply) from an Inkbox mailbox.

Usage:
    INKBOX_API_KEY=ApiKey_... MAILBOX_ADDRESS=agent@inkboxmail.com python send_email.py
"""

import os
from inkbox.mail import InkboxMail

client = InkboxMail(api_key=os.environ["INKBOX_API_KEY"])
mailbox_address = os.environ["MAILBOX_ADDRESS"]

# Send a new email
sent = client.messages.send(
    mailbox_address,
    to=["recipient@example.com"],
    subject="Hello from Inkbox",
    body_text="Hi there! This message was sent via the Inkbox SDK.",
    body_html="<p>Hi there! This message was sent via the <strong>Inkbox SDK</strong>.</p>",
)
print(f"Sent message {sent.id}  subject={sent.subject!r}")

# Reply to that message (threads it automatically)
reply = client.messages.send(
    mailbox_address,
    to=["recipient@example.com"],
    subject=f"Re: {sent.subject}",
    body_text="Following up on my previous message.",
    in_reply_to_message_id=str(sent.id),
)
print(f"Sent reply {reply.id}")
